src/IECore/IECore.cpp

```diff
@@ -55,12 +60,27 @@ int patchVersion()
 	return IE_CORE_PATCHVERSION;
 }
 
+int compatibilityVersion()
+{
+	return IE_CORE_MILESTONEVERSION * 1000 + IE_CORE_MAJORVERSION;
+}
+
+const std::string &compatibilityVersionString()
+{
+	static std::string v;
+	if( !v.size() )
+	{
+		v = boost::str( boost::format( "%d.%d" ) % milestoneVersion() % majorVersion() );
+	}
```

Comment:
This isn't thread safe, but I think doing the formatting during initialisation would be :

```suggestion
	static std::string v = boost::str( boost::format( "%d.%d" ) % milestoneVersion() % majorVersion() );
```

Reply:
Oh I just copied what we've been doing in `versionString()` below. I guess we should update both to the safe version.

Reply:
Fixed in 4a0b449

Reply:
4a0b4493b6327e7d52dbe6768438c315ba11a214 doesn't actually seem to be in this PR, so you may have received some separate notifications from when I commented on it after following the link. It looks like you've since added `inline`, which should have avoided the comment I made about duplicate symbols.

Reply:
Yeah, it was a battle with the Windows CI. I'd originally inlined them, but then removed it because I'd read something saying you shouldn't inline if they're rarely used, but then Windows insisted I inline if I was defining... anyhow if I move then all back to IECore.h as you suggested on the other comment then I'll stop inlining and define in the cpp as we had before.